### PR TITLE
Keep IDBD traces from turning 0*inf into NaN

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
#42 already skips the IDBD meta-update when `error * x * h` is non-finite, so log step-sizes stay finite. The h-trace and weight delta still do `alpha * error * x`. A silent feature (`x=0`) against an inf error is `0 * inf` = NaN, and that channel stays poisoned through later finite updates. Nonzero features still take an unbounded inf step.

NaN products are mapped to 0. Genuine infs stay inf. The existing log-alpha guard is unchanged.

Failing-test-first: inf error with a zero feature wrote NaN traces and weight deltas on main, then wrote 0 on that channel. `tests/test_optimizers.py`: 52 passed.

Follow-on to #42 (meta-update). Independent of #56 (Autostep) and #61 (ObGD bounder).

No Slop run receipt: this session is Cursor Grok, not Codex `gpt-5.6-sol` or Claude Code `claude-fable-5`.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by deepanshu-yd — CHANGES_REQUESTED** [review]
Blocking: blanket `jnp.where(jnp.isnan(h_step), 0, h_step)` (and the same on `weight_delta` / `trace_step`) in `update` and `update_from_gradient`.

This does fix `0 * inf` on a silent feature. It also rewrites a **genuine NaN error** (or NaN alpha) on a **nonzero** feature into a zero update, so the next finite step looks healthy. That is the defect 0xcii blocked on #61 and that #69 already closed with `_zero_if_zero_times_inf` (NaN **and** one operand inf, the other exactly 0).

Please copy that predicate and add a negative test: NaN error, `x = [1, 0]` → channel 0 stays NaN, not 0.

#72 and #73 edit the same `optimizers.py` from this same base; they will conflict. Maintainer sequencing, not a reason to keep the mask.

```slop-review
{"schemaVersion":"1","projectId":"asi","artifactUrl":"https://github.com/elizaOS/asi/pull/70","headSha":"89e6242521daeaeb0531b683693063fea6862497","recommendation":"reject","reproduced":false,"securityRisk":"none","duplicateRisk":"none","usefulArtifacts":[],"commands":[],"evidenceUrls":[],"summary":"Blanket where(isnan) on IDBD traces/deltas zeros genuine NaNs. Needs the 0*inf-only gate from #61/#69."}
```

AI provider/model: xAI / grok-4.6
Client / agent tooling: grok CLI
Live execution of the untrusted head is blocked on this host. Static review of the raw diff against `main` `da8b86c`.
— [deepanshu-yd]

**Comment by lalalune** [comment]
Closing as rejected after the Alberta 0.29 review, not as merged. This patch broadly rewrites NaN products to zero and therefore cannot distinguish the intended 0×∞ case from a genuine upstream NaN. That masks numerical corruption as a valid-looking no-op and weakens fail-closed measurement behavior. The existing requested-changes review remains the audit rationale; a future fix would need to gate on the exact collapsed-bound condition or reject the non-finite input without erasing genuine NaNs.
